### PR TITLE
BLD: bump numpy min version to 1.15.4

### LINKED ISSUE #33718: DEP: bump numpy min version?
We currently require 1.13.3 which was released 2017-09-29.  The last bump was in 0.25.0 (#25554).

I think its time for another bump.  The release dates for the 1.x.0 and 1.x.MAX versions out there:

1.14.0 -> 2018-01-06
1.14.6 -> 2018-09-23
1.15.0 -> 2018-07-23
1.15.4 -> 2018-11-04
1.16.0 -> 2019-01-13
1.16.6 -> 2019-12-29
1.17.0 -> 2019-07-06
1.17.5 -> 2020-01-01

1.15.4 looks not-unreasonable to me.

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #33718
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.1.0.rst [hunk 1]
```diff
@@ -382,6 +382,74 @@ Assignment to multiple columns of a :class:`DataFrame` when some of the columns
...
+If installed, we now require:
+
++-----------------+-----------------+----------+---------+
+| Package         | Minimum Version | Required | Changed |
++=================+=================+==========+=========+
+| numpy           | 1.15.4          |    X     |    X    |
++-----------------+-----------------+----------+---------+
+| pytz            | 2015.4          |    X     |         |
++-----------------+-----------------+----------+---------+
+| python-dateutil | 2.6.1           |    X     |         |
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.0.rst:400]: we had a PR which bumps dateutil, I *thought* we merged this already

Comment (jbrockmendel) [reply doc/source/whatsnew/v1.1.0.rst:400]: #33363 i think, will update this table

**Comment by jbrockmendel** [comment]
Mothballing to clear the queue

File: doc/source/whatsnew/v1.1.0.rst [hunk 2]
```diff
@@ -156,13 +156,23 @@ Other enhancements
...
+Some minimum supported versions of dependencies were updated (:issue:`33718`, :issue:`29766`, :issue:`29723`, pytables >= 3.4.3).
 If installed, we now require:
 
 +-----------------+-----------------+----------+---------+
 | Package         | Minimum Version | Required | Changed |
 +=================+=================+==========+=========+
-| python-dateutil | 2.7.3           |    X     |         |
+| numpy           | 1.15.4          |    X     |    X    |
++-----------------+-----------------+----------+---------+
+| pytz            | 2015.4          |    X     |         |
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.0.rst:167]: we should prob bump this just for consistency

Comment (jbrockmendel) [reply doc/source/whatsnew/v1.1.0.rst:167]: sure.  what to?

**Comment by jbrockmendel** [comment]
@WillAyd the MacOSPy36 failure is at build-time, any thoughts?

@TomAugspurger the LinuxPY36 failure looks like its a conda-dependency issue, any thoughts?

One of the other two failures should be fixed by #33898 leaving one more that i need to troubleshoot.

**Comment by TomAugspurger** [comment]
Seems to be an issue between NumPy 1.15.4 and python=3.6.1

```
 ~  conda create -n ttttt -c defaults -c conda-forge numpy=1.15.4 python=3.6.1                                                                                                                                                                                                                                                           10.9s  Thu Apr 30 16:20:32 2020
Collecting package metadata (current_repodata.json): done
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: |
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versions

Package python conflicts for:
python=3.6.1
numpy=1.15.4 -> python[version='>=2.7,<2.8.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0']
numpy=1.15.4 -> mkl_fft[version='>=1.0.6,<2.0a0'] -> python[version='2.7.*|3.5.*|3.6.*|>=3.5,<3.6.0a0|>=3.8,<3.9.0a0']
Note that strict channel priority may have removed packages required for satisfiability.
```

Python 3.6.2 seems fine. I'd recommend removing the `defaults` channel from that build.

**Comment by WillAyd** [comment]
Haven't debugged macOS failure but one thing worth updating first perhaps in the dependency file for it. I think it still has numpy 1.14 pinned

https://github.com/pandas-dev/pandas/blob/1940fcea056325cbe878c72078c51f80ea8482f4/ci/deps/azure-macos-36.yaml#L22

**Comment by jbrockmendel** [comment]
> Python 3.6.2 seems fine. I'd recommend removing the defaults channel from that build.

first couple tries didnt work, any other suggestions?  no rush

**Comment by TomAugspurger** [comment]
If you're able to reproduce locally, then binary search on the env file is often best (comment out half the file, `conda env create -f ...`, see if it breaks).

Turns out scipy=0.19.0 was the (or at least a) problem. It's not compatible with NumPy 1.15.4.

I'd recommend bumping scipy to 1.2.0, which is the first release after NumPy 1.15.4.

**Comment by TomAugspurger** [comment]
Can you bump the versions in `pandas/compat/_optional.py` for scipy and any other optional deps that are increased?

**Comment by jbrockmendel** [comment]
> I'd recommend bumping scipy to 1.2.0, which is the first release after NumPy 1.15.4.

that did the trick, thanks

File: doc/source/whatsnew/v1.1.0.rst (hunk 2 above)
Comment (jreback) [reply doc/source/whatsnew/v1.1.0.rst:168]: i guess 2019 ish, but maybe doesn't matter

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
great!

File: pyproject.toml [hunk 3]
```diff
@@ -5,8 +5,8 @@ requires = [
     "setuptools",
     "wheel",
     "Cython>=0.29.16",  # Note: sync with setup.py
-    "numpy==1.13.3; python_version=='3.6' and platform_system!='AIX'",
-    "numpy==1.14.5; python_version>='3.7' and platform_system!='AIX'",
+    "numpy==1.15.4; python_version=='3.6' and platform_system!='AIX'",
+    "numpy==1.15.4; python_version>='3.7' and platform_system!='AIX'",
```
Comment (olt) [inline pyproject.toml:9]: Shouldn't that be "numpy>=1.15.4"? I had issues installing pandas from source with numpy 1.18.5 installed.
Should I create a ticket for that?

Comment (TomAugspurger) [reply pyproject.toml:9]: I don't believe so. We build with the oldest supported NumPy that is compatible with the version of NumPy.

> I had issues installing pandas from source with numpy 1.18.5 installed.

This also doesn't quite make sense. The idea behind pyproject.toml is to isolate the build environment. It shouldn't matter what version of NumPy was used to build pandas with, as long as it's binary compatible (IIUC).

Comment (olt) [reply pyproject.toml:9]: Ok, ignore my comment if that is how you make sure that your build with the oldest supported version.

In my case I have numpy 1.18.5 installed and I want to use it to build pandas from source. However, it complains that the build requirements are note met. 
```
Installing build dependencies ... error
ERROR: Complete output from command /opt/project/venv37/bin/python3.7 /opt/project/venv37/lib/python3.7/site-packages/pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-0xz070yu/overlay --no-warn-script-location --no-binary :none: --only-binary :none: --no-index --find-links https://project:omniscale@pkgs.omniscale.net:443/project -- setuptools wheel 'Cython>=0.29.13' 'numpy==1.13.3; python_version=='"'"'3.6'"'"' and platform_system!='"'"'AIX'"'"'' 'numpy==1.14.5; python_version>='"'"'3.7'"'"' and platform_system!='"'"'AIX'"'"'' 'numpy==1.16.0; python_version=='"'"'3.6'"'"' and platform_system=='"'"'AIX'"'"'' 'numpy==1.16.0; python_version>='"'"'3.7'"'"' and platform_system=='"'"'AIX'"'"'':
ERROR: Ignoring numpy: markers 'python_version == "3.6" and platform_system != "AIX"' don't match your environment
Ignoring numpy: markers 'python_version == "3.6" and platform_system == "AIX"' don't match your environment
Ignoring numpy: markers 'python_version >= "3.7" and platform_system == "AIX"' don't match your environment
Looking in links: https://*redacted*
Collecting setuptools
Collecting wheel
Collecting Cython>=0.29.13
Collecting numpy==1.14.5
  ERROR: Could not find a version that satisfies the requirement numpy==1.14.5 (from versions: 1.18.5)
ERROR: No matching distribution found for numpy==1.14.5
```

Comment (TomAugspurger) [reply pyproject.toml:9]: However, were you using python 3.8? There was an issue with us not having a line for py38. https://github.com/pandas-dev/pandas/pull/34667

Comment (olt) [reply pyproject.toml:9]: No, 3.7.

Comment (olt) [reply pyproject.toml:9]: Ok, I think I didn't quite understood how build isolation works. Thanks for your quick relies!

**Pull request merged**